tests/test_git_init.py

```diff
@@ -2,49 +2,31 @@
 
 import pathlib
 import subprocess
+from typing import Callable
 
 import pytest
 
 
 @pytest.mark.parametrize(
-    "git_init_cookiecutter_option",
-    ["initialise_git_repository=True", "initialise_git_repository=False"],
+    "git_init",
+    [{"initialise_git_repository": True}, {"initialise_git_repository": False}],
 )
 def test_initialisation_of_git_repo(
+    generate_package: Callable,
     tmp_path: pathlib.Path,
-    project_config: dict,
-    git_init_cookiecutter_option: str,
+    git_init: dict[str, bool],
 ) -> None:
-    """
-    Checks to see if git was correctly initialised if desired.
-
-    Args:
-        tmp_path: A temporary directory path object which is unique.
-        project_config: A dictionary with values for the cookiecutter template,
-            as defined in the cookiecutter.json
-        git_init_cookiecutter_option: A string defined in cookiecutter.json,
-            which determines if git should be initialised or not
-
-    """
+    """Checks to see if git was correctly initialised if desired."""
+    test_config = {
+        "github_username": "test-user",
+        "project_short_description": "description",
+        "project_name": "Cookiecutter Test",
+        **git_init,
+    }
     # Run cookiecutter with initialise_git_repository
-    cookie = subprocess.run(
-        [  # noqa: S603,S607
-            "cookiecutter",
-            ".",
-            "--no-input",
-            "--output-dir",
-            f"{tmp_path}",
-            f"github_username={project_config['github_username']}",
-            f"project_name={project_config['project_name']}",
-            f"project_short_description={project_config['project_short_description']}",
-            git_init_cookiecutter_option,
-        ],
-        capture_output=True,
-        check=True,
-        text=True,
-    )
+    result = generate_package(config=test_config, path=tmp_path)
 
-    test_project_dir = tmp_path / project_config["expected_repo_name"]
+    test_project_dir = tmp_path / "cookiecutter-test"
```

Comment:
Would it be better to have that in `test_config`? Feels a bit hard-codey

Reply:
Oh yeah. David changed it. Hmm... I don't really mind. We no longer have the "expected" keys in the config dictionary, because it's really just the input. Perhaps this way is clearer?